### DISCUSSION TITLE
The request URL was not found on the server

### DISCUSSION THREAD

**Discussion by dedalusmasken** [rictom/rede-cnpj#16] [Q&A]
Fala, pessoal,
Estou tentando rodar o rede.py e encontro um erro na abertura do navegador, aonde ele não consegue carregar a página local. 
O endereço está no 127.0.0.1:5000/rede e recebo uma mensagem de erro:

{"message": "The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again."}

Obrigado!

**Comment by rictom** [comment]
Olha, você precisaria detalhar, qual o comanda que vc fez no console, copia do texto do console.  Primeiro, vc tem que estar no console do Anaconda, daí todas as linhas começam como
(base) C:\pasta_atual...

**Reply by dedalusmasken** [reply]
Ele só roda no Anaconda? Eu estava tentando rodar direto no console. Segue console:

C:\Users\dedalusmasken\Documents\SCRIPT\PYTHON\REDE CNPJ\rede-cnpj-master\rede>python rede.py
Carregando rede_sqlite_cnpj
Utilizando rede_sqlite_cnpj como rede_relacionamentos.
C:\Users\dedalusmasken\AppData\Roaming\Python\Python311\site-packages\flask_limiter\extension.py:293: UserWarning: Using the in-memory storage for tracking rate limits as no storage was explicitly specified. This is not recommended for production use. See: https://flask-limiter.readthedocs.io#configuring-a-storage-backend for documentation about configuring the storage backend.
  warnings.warn(
 * Serving Flask app 'rede'
 * Debug mode: on
WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:5000
 * Running on http://XXXXXXX:5000

**Comment by rictom** [comment]
é um detalhe... Você precisar rodar o console no "ambiente" do Anaconda. Tem que abrir o console pelo botão iniciar do Windows 
![image](https://github.com/rictom/rede-cnpj/assets/71139693/e3d94150-4f48-4577-9c23-d76b7e29c11a)

**Reply by dedalusmasken** [reply]
então, consegui rodar pelo Anaconda, mas acabei encontrando o mesmo problema:

![image](https://github.com/rictom/rede-cnpj/assets/133978128/e627124f-656d-4bd8-88c7-9dbc41a33965)


E a mensagem, a princípio, está OK no console

(base) C:\Users\dedalusmasken\Documents\SCRIPT\PYTHON\REDE CNPJ\rede-cnpj-master\rede>python rede.py
Carregando rede_sqlite_cnpj
Utilizando rede_sqlite_cnpj como rede_relacionamentos.
C:\Users\dedalusmasken\AppData\Roaming\Python\Python37\site-packages\flask_limiter\extension.py:294: UserWarning: Using the in-memory storage for tracking rate limits as no storage was explicitly specified. This is not recommended for production use. See: https://flask-limiter.readthedocs.io#configuring-a-storage-backend for documentation about configuring the storage backend.
  "Using the in-memory storage for tracking rate limits as no storage "
 * Serving Flask app "rede" (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: on
 * Running on http://0.0.0.0:5000/ (Press CTRL+C to quit)

**Comment by rictom** [comment]
Essa última linha indica o endereço na rede local. Nesse caso teria que por  http://0.0.0.0:5000/rede no navegaor para vc ver a rede. Mas não entendi porque apareceu 0.0.0.0....

**Comment by rictom** [comment]
No final do arquivo rede.py, tem um codigo:
if __name__ == '__main__':
    import webbrowser, platform
    if platform.system()=='Windows': 
        webbrowser.open(f'http://127.0.0.1:{config.par.porta_flask}/rede', new=0, autoraise=True) 
    app.run(host='0.0.0.0',debug=True, use_reloader=False, port=config.par.porta_flask)

Essa linha host='0.0.0.0' seria pra que a aplicação recebesse pedidos de outras máquinas na rede. Mas na hora que roda, o meu console aparece algo como 
WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:5000
 
Eu sugiro vc remover essa parte do 0.0.0.0,
colocando app.run(debug=True, use_reloader=False, port=config.par.porta_flask) e ver o que acontece

**Reply by dedalusmasken** [reply]
Isso resolveu a mensagem no console

" Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)"

Mas ainda não consigo visualizar no navegador. Vou seguir investigando aqui.

**Comment by rictom** [comment]
Olha, as vezes se vc colocar só 127.0.0.1 no navegador, ele está achando que é https;//127.0.0.1. 
Cole no navegador http://127.0.01:5000/rede
Se não funcionar, tente noutro navegador
